02_senbei/2021_02/main.rb

```diff
@@ -0,0 +1,137 @@
+def main
+    int.times do
+        n, m = intary
+        ans = []
+        # 最小連結成分数
+        min = n - m
+        min = 1 if min <= 0
+        ans << min
+
+        # 最大連結成分数
+        min = 0
+        max = 1000000000
+        while min + 1 != max
+            center = (min + max) / 2
+            ruiseki = 0.5 * center * (center + 1)
```

Comment:
最初に見落としていましたが、できれば整数型を徹底して置いた方が安全かなー。
実害はないと思うのですが、習慣としておくと良いと思っています。

```suggestion
            ruiseki = center * (center + 1) / 2
```